### PR TITLE
add support for saf key for aes encryption

### PR DESCRIPTION
**Pull request by acdemyers**
Adding code to allow command line support for getting an AES key from a saf keyring.

### CODE REVIEW

**Comment by acdemyers** [comment]
#build

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops [hunk 1]
```diff
@@ -86,6 +86,16 @@ encode.option-desc.custom.description=\
 # encryption is enabled.
 no.custom.encyption=The custom password encryption is not detected.
 
+# The following messages are for retrieving an encryption key from a 
+# SAF keyring.
+saf.label.does.not.exist=The {0} label does not exist in the {1} keyring.  Unable to find a valid key for AES password encryption.
+saf.cert.is.not.key=The {0} certificate is not a key entry.  The certificate needs to be a key entry for use as an AES password encryption key.
+saf.cert.expired=The {0} certificate is expired and cannot be used as an AES password encryption key.
+saf.keyring.does.not.exist=The {0} keystore does not exist.  No certificate can be retrieved from the keystore to use as the AES password encryption key.
+saf.arg.not.onZ=The --keyring, --keyringType, and --keyLabel arguments can only be used on z/OS.
```
Comment (dmuelle) [inline dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops]: suggested edit:

The --keyring, --keyringType, and --keyLabel arguments can only be used on z/OS.
--->
The --keyring, --keyringType, and --keyLabel arguments can be used only on z/OS.

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops [hunk 2]
```diff
@@ -86,6 +86,16 @@ encode.option-desc.custom.description=\
 # encryption is enabled.
 no.custom.encyption=The custom password encryption is not detected.
 
+# The following messages are for retrieving an encryption key from a 
+# SAF keyring.
+saf.label.does.not.exist=The {0} label does not exist in the {1} keyring.  Unable to find a valid key for AES password encryption.
+saf.cert.is.not.key=The {0} certificate is not a key entry.  The certificate needs to be a key entry for use as an AES password encryption key.
```
Comment (dmuelle) [inline dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops]: suggested edit

The certificate needs to be a key entry for use as an AES password encryption key.
--->
The certificate must be a key entry to use it as an AES password encryption key.

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops [hunk 3]
```diff
@@ -86,6 +86,16 @@ encode.option-desc.custom.description=\
 # encryption is enabled.
 no.custom.encyption=The custom password encryption is not detected.
 
+# The following messages are for retrieving an encryption key from a
```
Comment (dmuelle) [inline dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops:89]: for all messages in this set, only need one space after a period.

**Review by dmuelle — CHANGES_REQUESTED** [review]
just a few minor suggestions. Unless there is a configuration-related reason to use two spaces after a period between two sentences, only need one space after period.

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops (hunk 1 above)
Comment (acdemyers) [reply dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops]: The message has been updated please review.

File: dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops (hunk 3 above)
Comment (acdemyers) [reply dev/com.ibm.ws.security.utility/resources/com/ibm/ws/security/utility/resources/UtilityMessages.nlsprops:89]: I've updated the double spaces in that section,  please review.

**Review by dmuelle — APPROVED** [review]
LGTM Thanks

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Review by awisniew90 — APPROVED** [review]
LGTM

**Comment by acdemyers** [comment]
#libby

**Pull request merged**